inst/tutorials/vaccines/vaccines.Rmd

```diff
@@ -0,0 +1,332 @@
+                            method = "loess") 
+  lines(comparison$time_year, predict(curve_valuescomp), 
+        col = cols[6], lwd = 3)
+  legend("topright", box.lty = 0, bg = "white",
+         legend = c("Unsmoothed incidence for\nvaccine scenario",
+                    "Smoothed incidence for\nvaccine scenario",
+                    "Smoothed incidence for no\nintervention scenario"),
+         col = c(cols[2], cols[5], cols[6]), lty = c(1, 1, 1), lwd = 2.5, cex = 0.8, y.intersp = 1.5)
+}
+
+# Plot parasite prevalence
+plot_prev <- function(output){
+
+    comparison <- output_control
+    vaccinetime <- 1
+  
+  output$time_year <- output$timestep / year
+  comparison$time_year <- comparison$timestep / year
+
+  plot(x = output$time_year, y = output$n_detect_730_3650/output$n_730_3650, 
+       type = "l", col = cols[3], ylim=c(0,1), lwd = 3,
+       xlab = "Time (years)", ylab = expression(paste(italic(Pf),"PR"[2-10])),
+       xaxs = "i", yaxs = "i")
+  grid(lty = 2, col = "grey80", lwd = 0.5)
+  lines(x = comparison$time_year, y = comparison$n_detect_730_3650/comparison$n_730_3650,
+        col = cols[6], lwd = 3)
+  abline(v = vaccinetime, col = "black", lty = 2, lwd = 2.5)
+  text(x = vaccinetime + 0.05, y = 0.9, labels = "Start of\nvaccination", adj = 0, cex = 0.8)
+  legend("topright", box.lty = 0, 
+         legend = c("Prevalence for\nvaccine scenario",
+                    "Prevalence for no\nintervention scenario"),
+         col = c(cols[3], cols[6]), lty = c(1, 1), 
+         lwd = 2.5, cex = 0.8, y.intersp = 1.5)
+}
+
+# Plot dose timing
+plot_doses <- function(output){
+  output$month <- ceiling(output$timestep / month)
+  doses <- output[, c(grep("n_pev" , names(output)), grep("month", names(output)))]
+  doses <- aggregate(cbind(doses[1:4]),
+                      by = list(doses$month), 
+                      FUN = sum)
+  doses <- as.matrix(t(doses[, -1]))
+  
+  barplot(doses, xlab = "Month",
+          ylab = "Number of doses",
+          col = cols[1:6], space = 0, axes = T,
+          beside = FALSE, xaxs = "i", yaxs = "i",
+          ylim = c(0, max(colSums(doses))*1.1))
+  grid(lty = 2, col = "grey80", lwd = 0.5);box()
+  axis(side = 1, lty = 1, col = "black", pos = 0)
+  legend("topleft", box.lty = 0, legend = c("Dose 1","Dose 2","Dose 3","Booster 1"),
+         fill = cols[1:6], bg="transparent", cex = 0.8, y.intersp = 1.5)
+}
+
+# Control parameters and model output
+sim_length <- 3 * year
+human_population <- 10000
+starting_EIR <- 20
+
+simparams <- get_parameters(list(
+    human_population = human_population,
+    clinical_incidence_rendering_min_ages = 0,
+    clinical_incidence_rendering_max_ages = 5 * year,
+    individual_mosquitoes = FALSE
+  )
+)
+
+simparams <- set_equilibrium(parameters = simparams, init_EIR = starting_EIR)
+
+# Run a model with no interventions in a setting with no seasonality
+output_control <- run_simulation(timesteps = sim_length * 2, parameters = simparams)
```

Comment:
This line makes the intial set up of the doc very slow. I don't think it is needed as the user runs a control run themseleves early on

Reply:
I've tidied up the set up chunk a bit and removed the library(data.table) call. Currently it's taking the exercise 10-15secs to load which I think is acceptable-ish? I'm going through the essential things now but if I get time I can move the baseline scenario to each of the chunks and just have it re-run when needed which will speed things up.

Reply:
Have no moved the control/no vaccine runs from the set-up chunk to the exercise chunks along with the code to generate the plots (puled outside of a function) - makes the chunks a bit bulkier but I think it's helpful to give them all of the code in each case and avoid any ambiguity.